components/Forums/ThreadCommentCard.js

```diff
@@ -0,0 +1,240 @@
+// Module imports
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import Cookies from 'js-cookie'
+import FontAwesomeIcon from '@fortawesome/react-fontawesome'
+import moment from 'moment'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+
+
+
+// Component imports
+import { actions } from '../../store'
+import Avatar from '../Avatar'
+import Component from '../Component'
+
+class ThreadCommentCard extends Component {
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  async _handleRemoveButtonClick (event) {
+    const {
+      name,
+    } = event.target
+
+    const {
+      confirmingRemove,
+    } = this.state
+
+    const {
+      deleteThreadComment,
+      onDelete,
+      comment,
+    } = this.props
+
+    if (confirmingRemove && name === 'confirm') {
+      this.setState({ confirmingRemove: false, removing: true })
+
+      const { status } = await deleteThreadComment(comment.id)
+
+      if (status === 'success') {
+        this.setState({ removing: false, removed: true })
+
+        if (typeof onDelete === 'function') {
+          onDelete(comment)
+        }
+      } else {
+        this.setState({ removing: false })
+      }
+    } else if (confirmingRemove && name === 'deny') {
+      this.setState({ confirmingRemove: false })
+    } else {
+      this.setState({ confirmingRemove: true })
+    }
+  }
+
+  /***************************************************************************\
+    Public Methods
+  \***************************************************************************/
+
+  async componentDidMount () {
+    let { user } = this.props
+
+    if (!user) {
+      const { payload, status } = await this.props.getUser(this.props.posterId)
```

Comment:
Shouldn't we do this in `getInitialProps`?

Reply:
This should hardly ever get called as the user is included with the thread GET. This is mainly a fallback in case the user is not in the store for whatever reason.

there SHOULD be a getInitialProps in list, but I would need to rethink how the thread reducers in the store works for that to work properly.

It's all kinda hairy because of pagination. It might ACTUALLY be easier to do an infinite scrolling list page :P

Reply:
Update: This will be handled in the page's getInitalProps once we better support pagination in the store.